otherlibs/dune-rpc/dune_rpc.mli

```diff
@@ -173,8 +210,9 @@ module V1 : sig
 
       val create :
            ?log:(Message.t -> unit Fiber.t)
-        -> ?errors:(Error.t list -> unit Fiber.t)
-        -> ?promotions:(Promotion.t list -> unit Fiber.t)
+        -> ?diagnostic:(Diagnostic.Event.t -> unit Fiber.t)
```

Comment:
If a client connects to a build midway through, will they only receive errors from that point onwards? Presumably, they can get the initial diagnostic state via the `diagnostics` request, but it might be friendly to also give this information to the handler.

Reply:
> Presumably, they can get the initial diagnostic state via the diagnostics request

Yeah, that is how I expected it to work.

>  but it might be friendly to also give this information to the handler.

How would this information be given?

Reply:
One way might be to adjust handlers to include `initial_diagnostics_state:(Diagnostic.t list -> unit Fiber.t)`, and then the server sends that information when the session is first invoked.

That said, this isn't a crucial feature by any means.

Reply:
I'd rather not have two ways of doing the same thing unless there's clear trade-offs between the 2 mechanisms.

Reply:
The race is quite problematic. Then I suppose I'll change the subscription mechanism as you advised, and get rid of the request. I don't want to add requests unless they are strictly useful to implement a feature.

Reply:
> One way might be to adjust handlers to include initial_diagnostics_state:(Diagnostic.t list -> unit Fiber.t), and then the server sends that information when the session is first invoked.

Actually, I'm wondering if we should just reuse the exiting callback:
```ocaml
...
        -> ?diagnostic:(Diagnostic.Event.t list -> unit Fiber.t)
...
```

And we'll guarantee that initially we'll just initially send the initial subscriptions as `Add err` events.

Reply:
Sure, that seems like a reasonable idea.